### PR TITLE
Feature test

### PR DESCRIPTION
**Pull request by codlab**
Using latest AS versions :
- switch test artifacts to : Unit Tests
- removed testInstrumentationRunner (using default test runner)
- fix missing 'src/test/resources' dependency
- restore log4j12

This should be more elegant than the added workaround which is no longer valid ! (fixed in gradle 1.2+)

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -9,13 +8,17 @@ android {
...
 
     sourceSets {
         main {
             java.srcDirs = ['src/main/java-gen', 'src/main/java']
         }
+
+        //make AS and Gradle aware of the active dependencies
+        androidTest {
+            java.srcDirs = ['src/main/java-gen', 'src/test/java', 'src/test/resources']
+        }
```
Comment (afwang) [inline app/build.gradle]: Granted that I'm still very new to Gradle build systems, why should androidTest be using the src/test directory? My understanding is that the src/test directory is meant for local unit tests meant to be run on the host machine, while the src/androidTest directory is meant for unit tests to run on a device. Please feel free to correct me if I have an error in my understanding.

**Comment by afwang** [comment]
I'll give this a try when I get home. I'm pretty sure I'm on the latest version of AS, but I was running into the test resources issue yesterday. I'll double-check later.

File: app/build.gradle (hunk 1 above)
Comment (codlab) [inline app/build.gradle]: +1 typo here

the correct should be here:

        androidTest {
            java.srcDirs = ['src/main/java-gen']
        }

due to an issue which existed on pre-1.2.3 gradle support (it was presumably fixed in AS but I am currently running the new RC and did not inspected it)

Also note that the current tests in the Android Instrumentation Folder are Unit Test

**Comment by codlab** [comment]
PR updated with parsing fixes, code cleaned and test updated with new model representation suggested.

PR name is innacurate and impact : tests / ui / parsers / in app data model

**Comment by afwang** [comment]
I see the pull parser being replaced with JSoup. I'm very wary of DOM parsing since the last time I tried it, I was waiting 15-20 seconds before events were showing up on screen. I usually wait right around 10 seconds with the pull parser implementation.

**Comment by codlab** [comment]
What versions did you used ? 1.8.+ improved performances (1.8.2 also introduced parallel parsing). It is way better now. XmlPullParser is the kind of library to use when you are using a lexer.

- maintenance & code quality improved
- performance are equals

**Comment by afwang** [comment]
The version I used was version JSoup 1.8.1. I can give this version a try. Let's start documenting some benchmark results so we have a clearer picture of performance.

**Pull request merged**